src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -1129,21 +1130,23 @@ private static unsafe nuint UnalignedCountVector128(ref byte searchSpace)
 
         public static void Reverse(ref byte buf, nuint length)
         {
-            if (Avx2.IsSupported && (nuint)Vector256<byte>.Count * 2 <= length)
+            Debug.Assert(length > 1);
+
+            nint remainder = (nint)length;
+            nint offset = 0;
+
+            if (Avx2.IsSupported && remainder >= Vector256<byte>.Count)
```

Comment:
The trick with taking the vectorized path for sizes smaller than 2*Vector256<byte>.Count is neat, but it hurts performance in number of cases. There can be a lot of redundant work done for certain sizes with this PR. 

I have opened https://github.com/dotnet/runtime/issues/78604 on this.